src/main/java/com/algorand/algosdk/transaction/Transaction.java

```diff
@@ -78,15 +122,219 @@
+            return new SignedTransaction(this, txSig, txID);
+        } catch (IOException e) {
+            throw new RuntimeException("unexpected behavior", e);
+        }
+    }
+
+    /**
+     * Populate the fee with suggestedFeePerByte * estimateTxSize.
+     * @param suggestedFeePerByte suggestedFee given by network
+     * @throws NoSuchAlgorithmException could not estimate tx encoded size.
+     */
+    public void setFeeWithSuggestedFeePerByte(BigInteger suggestedFeePerByte) throws NoSuchAlgorithmException{
+        BigInteger newFee = suggestedFeePerByte.multiply(this.estimatedEncodedSize());
+        this.setFee(newFee);
+    }
+
+    /**
+     * EstimateEncodedSize returns the estimated encoded size of the transaction including the signature.
+     * This function is used for calculating the fee from suggested fee per byte.
+     * @return an estimated byte size for the transaction.
+     */
+    public BigInteger estimatedEncodedSize() throws NoSuchAlgorithmException {
+        try {
+            return BigInteger.valueOf(
+                    Encoder.encodeToMsgPack(
+                            this.signTransaction(
+                                    SignatureUtils.getDummyPrivateKey().getPrivate())).length);
+        } catch (IOException e) {
+            throw new RuntimeException("unexpected behavior", e);
+        }
+    }
+
+    /**
+     * Set the transaction fee value.
+     * @param newFee the fee suggested for the transaction. If the
+     * newFee is less than the minimum transaction fee, it is replaced
+     * by the minimum fee. 
+     **/
+    public void setFee(BigInteger newFee) {
+        if (newFee.compareTo(MIN_TX_FEE_UALGOS) < 0) {
+            newFee = MIN_TX_FEE_UALGOS;
+        }
+        fee = newFee;
     }
 
-    public Transaction(Address fromAddr, Address toAddr, BigInteger fee, BigInteger amount, BigInteger firstRound,
-                       BigInteger lastRound,
-                       String genesisID, Digest genesisHash) {
-        this(fromAddr, fee, firstRound, lastRound, null, amount, toAddr, genesisID, genesisHash);
+    /**
+     * Creates a tx to mark the account as willing to accept the asset.
+     * @param acceptingAccount is a checksummed, human-readable address that
+     * will accept receiving the asset.
+     * @param flatFee is the transaction flat fee
+     * @param firstRound is the first round this txn is valid (txn semantics
+     * unrelated to asset management)
+     * @param lastRound is the last round this txn is valid
+     * @param note
+     * @param genesisID corresponds to the id of the network
+     * @param genesisHash corresponds to the base64-encoded hash of the genesis
+     * of the network
+     * @param assetCreator is the address of the asset creator
+     * @param assetIndex is the asset index
+     **/
+    public static Transaction acceptAssetTransaction(Address acceptingAccount, //AssetTransaction
```

Comment:
Please consider naming it either by the verb `createAcceptAssetTransaction` or simply `acceptAsset`